### PR TITLE
[Doc] [SQL] Addes Hive metastore Parquet table conversion section

### PR DESCRIPTION
**Pull request by liancheng**
This PR adds a section about Hive metastore Parquet table conversion. It documents:
1. Schema reconciliation rules introduced in #5214 (see [this comment](https://github.com/apache/spark/pull/5188#issuecomment-86531248) in #5188)
2. Metadata refreshing requirement introduced in #5339

### CODE REVIEW

File: docs/sql-programming-guide.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ The DataFrame API is available in [Scala](api/scala/index.html#org.apache.spark.
 All of the examples on this page use sample data included in the Spark distribution and can be run in the `spark-shell` or the `pyspark` shell.
 
 
-## Starting Point: `SQLContext`
+## Starting Point: SQLContext
```
Comment (liancheng) [inline docs/sql-programming-guide.md]: Removed the backquotes because the rendered text looks funny.

**Comment by liancheng** [comment]
Added Python snippet for metadata refreshing after adding `refreshTable` for PySpark  in #5349.

**Comment by yhuai** [comment]
I just found we have an example having the old code (https://spark.apache.org/docs/1.3.0/sql-programming-guide.html#programmatically-specifying-the-schema). We still have the following 
```
// Import Spark SQL data types and Row.
import org.apache.spark.sql._
```
Can you update it to use `import org.apache.spark.sql.types._` for data types?

**Comment by liancheng** [comment]
@yhuai This typo has been fixed in master and branch-1.3.

File: docs/sql-programming-guide.md [hunk 2]
```diff
@@ -1034,6 +1034,79 @@ df3.printSchema()
...
+   nullability.  The reconciled field should have the data type of the Parquet side, so that
+   nullability is respected.
+
+1. The reconciled schema contains exactly those fields defined in Hive metastore schema.
+
+   - Any fields that only appear in the Parquet schema are dropped in the reconciled schema.
+   - Any fileds that only appear in the Hive metastore schema are added as nullable field in the
+     reconciled schema.
+
+#### Metadata Refreshing
```
Comment (yhuai) [inline docs/sql-programming-guide.md]: Seems it is under `Hive metastore Parquet table conversion`. However, users may need to call `refresh table` in other cases, right? For example, when they manually copy data to the dir of a data source table.

File: docs/sql-programming-guide.md [hunk 3]
```diff
@@ -1034,6 +1034,79 @@ df3.printSchema()
...
+
+Spark SQL caches Parquet metadata for better performance.  When Hive metastore Parquet table
```
Comment (yhuai) [inline docs/sql-programming-guide.md]: Actually, I was wondering if we have a section to explain what is a data source table?

Comment (liancheng) [inline docs/sql-programming-guide.md]: Agree, missing such a section is part of the reason why I put the metadata refreshing section here...

**Comment by andrewor14** [comment]
@liancheng @yhuai what happened here? Do we still want to merge this?

**Comment by liancheng** [comment]
@yhuai I guess it's OK to merge this?

**Comment by yhuai** [comment]
@liancheng Yeah, it's fine to merge it. btw, you have added `refreshTable` in Python, right? We can update the description. Also, can you build the doc to double check the generated doc looks fine?

**Comment by liancheng** [comment]
@yhuai Finished updating Python `refreshTable` snippet and double checking rendered docs. Will merge this pending Jenkins.

**Comment by yhuai** [comment]
Sure. Thanks!

**Comment by andrewor14** [comment]
(you could just merge this since Jenkins doesn't actually test the docs)

**Comment by liancheng** [comment]
@andrewor14 Yeah... Thanks :) Merging to master.

**Pull request closed without merging**